src/Interactive/EditorFeatures/Core/Extensibility/Interactive/NuGetPackageResolverImpl.cs

```diff
@@ -0,0 +1,180 @@
+        {
+            using (var jsonWriter = new JsonTextWriter(writer))
+            {
+                jsonWriter.Formatting = Newtonsoft.Json.Formatting.Indented;
+                jsonWriter.WriteStartObject();
+                jsonWriter.WritePropertyName("dependencies");
+                jsonWriter.WriteStartObject();
+                jsonWriter.WritePropertyName(packageName, escape: true);
+                jsonWriter.WriteValue(packageVersion);
+                jsonWriter.WriteEndObject();
+                jsonWriter.WriteEndObject();
+            }
+        }
+
+        /// <summary>
+        /// Read the references from the project.lock.json file.
+        /// </summary>
+        internal static ImmutableArray<string> ReadProjectLockJson(string packagesDirectory, TextReader reader)
+        {
+            JObject obj;
+            using (var jsonReader = new JsonTextReader(reader))
+            {
+                obj = JObject.Load(jsonReader);
+            }
+            var builder = ArrayBuilder<string>.GetInstance();
+            var targets = (JObject)GetPropertyValue(obj, "targets");
+            // TODO: Choose the appropriate target. For now use the first.
+            foreach (var target in targets)
+            {
+                foreach (var package in (JObject)target.Value)
+                {
+                    var packageRoot = PathUtilities.CombineAbsoluteAndRelativePaths(packagesDirectory, package.Key);
+                    var runtime = (JObject)GetPropertyValue((JObject)package.Value, "runtime");
+                    if (runtime == null)
+                    {
+                        continue;
+                    }
+                    foreach (var item in runtime)
+                    {
+                        var path = PathUtilities.CombinePossiblyRelativeAndRelativePaths(packageRoot, item.Key);
+                        builder.Add(path);
+                    }
+                }
+                break;
+            }
+            return builder.ToImmutableAndFree();
+        }
+
+        private static JToken GetPropertyValue(JObject obj, string propertyName)
+        {
+            JToken value;
+            obj.TryGetValue(propertyName, out value);
+            return value;
+        }
+
+        private static void NuGetRestore(string projectJsonPath)
+        {
+            // Load nuget.exe from same directory as current assembly.
+            var nugetExePath = PathUtilities.CombineAbsoluteAndRelativePaths(
+                PathUtilities.GetDirectoryName(
+                    CorLightup.Desktop.GetAssemblyLocation(typeof(NuGetPackageResolverImpl).GetTypeInfo().Assembly)),
+                "nuget.exe");
+            var startInfo = new ProcessStartInfo()
+            {
+                FileName = nugetExePath,
+                Arguments = $"restore \"{projectJsonPath}\"",
+                CreateNoWindow = true,
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
```

Comment:
If you redirect StandardOutput and StandardError, you need to have something actively reading both. In your case, the StandardError pipe could block and then you'll deadlock.